sql/src/test/java/org/apache/druid/sql/calcite/CalciteInsertDmlTest.java

```diff
@@ -1435,7 +1436,10 @@ public void testErrorWithUnableToConstructColumnSignatureWithExtern()
                                 "general"
                             )
                                 .expectMessageContains(
-                                    "Cannot construct instance of `org.apache.druid.segment.column.ColumnSignature`, problem: `java.lang.NullPointerException`\n"
+                                    JvmUtils.majorVersion() >= 17
+                                    ? "Cannot construct instance of `org.apache.druid.segment.column.ColumnSignature`, problem: Cannot invoke \"String.length()\" because \"s\" is null"
+                                    :
+                                    "Cannot construct instance of `org.apache.druid.segment.column.ColumnSignature`, problem: `java.lang.NullPointerException`"
```

Comment:
Also can change this message contains to
```
Cannot construct instance of `org.apache.druid.segment.column.ColumnSignature
```
which I think would be same for all the JVM versions

Reply:
We can shorten the message as well. That will also work. I just want to highlight the fact that jackson is not bubbling up the generic NPR for jdk version >17 and giving specific root cause of the NPR exception.